features/steps/events.go

```diff
@@ -0,0 +1,69 @@
+package steps
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
+
+	kafka "github.com/ONSdigital/dp-kafka/v3"
+	"github.com/ONSdigital/dp-kafka/v3/kafkatest"
+	"github.com/ONSdigital/dp-search-reindex-tracker/event"
+	"github.com/ONSdigital/go-ns/avro"
+	"github.com/cucumber/godog"
+	"github.com/rdumont/assistdog"
+)
+
+func theseEventsAreConsumed[M event.KafkaAvroModel](c *SearchReindexTrackerComponent, schema *avro.Schema, consumer kafka.IConsumerGroup, table *godog.Table) error {
+	observationEvents, err := convertToEvents[M](table)
+	if err != nil {
+		return fmt.Errorf("failed to convert to events - err: %v", err)
+	}
+
+	//signals := registerInterrupt()
+
+	// consume extracted observations
+	for _, e := range observationEvents {
+		bytes, err := schema.Marshal(e)
+		if err != nil {
+			return fmt.Errorf("failed to marshal events - err: %v", err)
+		}
+
+		consumer.Channels().Upstream <- kafkatest.NewMessage(bytes, 0)
+	}
+
+	time.Sleep(300 * time.Millisecond)
+
+	// kill application
+	//signals <- os.Interrupt
```

Comment:
(minor): can remove this if it's not being used instead of commenting out ?

Reply:
Done